fix(celery): convert interval schedules to cron to survive beat restarts

webjunkie opened the pull request:
Celery beat resets interval-based schedule countdowns on restart. With beat pods restarting every 5-10 min, tasks with intervals longer than that never run - their countdown keeps resetting before reaching zero.

**Changes**
- Convert all interval schedules >= 60s to cron schedules (absolute, unaffected by restarts)
- Add warning to `add_periodic_task_with_expiry` docstring
- Add `PLAYLIST_COUNTER_SCHEDULE_CRON` setting for configurability

**Affected tasks:** warm_all_team_access_caches, ingestion_lag, clickhouse/pg metrics, process_scheduled_changes, replay_count_metrics, count_items_in_playlists, refresh_integrations

Example for task `warm_all_team_access_caches` shows it runs not as intended:
https://grafana.prod-us.posthog.dev/goto/46o_NLWvg?orgId=1

pauldambra commented on posthog/tasks/scheduled.py:
```diff
@@ -78,10 +76,12 @@ def add_periodic_task_with_expiry(
-    The expiry is larger than the schedule so that if the worker is only slightly delayed, it still gets processed.
+    Schedules a task with an expiry to prevent duplicate processing when workers fall behind.
+
+    IMPORTANT: Only use for short intervals (<60s). For longer intervals, use crontab schedules instead.
```
nobody reads comments :)
we should throw if the interval is longer than 60s

tbh we shojld have this only accept a crontab and convert all the usages

the point of this method was so that people wouldn't forget to add an expiry

if we're adding expiries manually we should fix or delete this :)

webjunkie replied:
I'm always up for being stricter 🫡

webjunkie committed:
refactor(celery): make helper accept crontab only, auto-calculate expiry

Address PR feedback:
- Helper now accepts crontab only (not intervals)
- Expiry defaults to 1.5x estimated interval, with optional override
- Added estimate_crontab_interval_seconds() to derive interval from
  crontab's expanded time sets (minutes, hours, days)
- Convert all existing call sites to use crontab schedules
- Keep short-interval tasks (<60s) using intervals directly since
  cron minimum resolution is 1 minute

github-actions[bot] committed:
Update query snapshots

pauldambra reviewed (approved):
nice
